calyx-py/calyx/builder.py

```diff
@@ -118,6 +118,14 @@ def input(self, name: str, size: int) -> ExprBuilder:
         self.component.inputs.append(ast.PortDef(ast.CompVar(name), size))
         return self.this()[name]
 
+    
+    def input_with_attributes(self, name: str, size: int, attribute_literals: List[Union[str, Tuple[str, int]]]) -> ExprBuilder:
```

Comment:
You might save yourself a little verbosity by creating a type alias:

```py
Attrs = List[Union[str, Tuple[str, int]]]
```

…and then using that in the function signatures.